cpp/src/generators/erdos_renyi_generator.cu

```diff
@@ -42,45 +41,38 @@ generate_erdos_renyi_graph_edgelist_gnp(raft::handle_t const& handle,
   CUGRAPH_EXPECTS(num_vertices < std::numeric_limits<int32_t>::max(),
                   "Implementation cannot support specified value");
 
-  auto random_iterator = thrust::make_transform_iterator(
-    thrust::make_counting_iterator<size_t>(0),
-    cuda::proclaim_return_type<float>([seed] __device__(size_t index) {
-      thrust::default_random_engine rng(seed);
-      thrust::uniform_real_distribution<float> dist(0.0, 1.0);
-      rng.discard(index);
-      return dist(rng);
-    }));
+  size_t max_num_edges = static_cast<size_t>(num_vertices) * num_vertices;
 
-  size_t count = thrust::count_if(handle.get_thrust_policy(),
-                                  random_iterator,
-                                  random_iterator + num_vertices * num_vertices,
-                                  [p] __device__(float prob) { return prob < p; });
-
-  rmm::device_uvector<size_t> indices_v(count, handle.get_stream());
+  auto generate_random_value = cuda::proclaim_return_type<float>([seed] __device__(size_t index) {
+    thrust::default_random_engine rng(seed);
```

Comment:
This might be out side the scope of this bug fix PR, but should we better use `raft::random::RngState`? Using two different random number generation mechanism does not sound ideal.

Reply:
Created an issue to address this.